amulet/deployer.py

```diff
@@ -330,6 +337,53 @@ def deploy_w_timeout_and_dir(self, timeout, deploy_dir):
             yield
         self.deployed = True
 
+    def action_defined(self, service):
+        """
+        :param service: Name of service to get list of defined actions for.
+
+        Returns the list of actions defined for the service.
+        """
+        if service not in self.services:
+            raise ValueError('Service needs to be added before listing actions.')
+        raw = juju(['action', 'defined', service, '--format', 'json'])
+        return json.loads(raw)
+
+    def action_do(self, unit, action, **kw):
```

Comment:
**kw doesn't work well in Juju because action parameters can contain `-` and other symbols Python doesn't like for key names. Instead a parameters dictionary would work better here, see the configure() action for an example